pkg/kubelet/status/status_manager.go

```diff
@@ -463,8 +467,26 @@ func (m *manager) TerminatePod(pod *v1.Pod) {
 		}
 	}
 
+	// Make sure all pods are transitioned to a terminal phase.
+	// TODO(#116484): Also assign terminal phase to static an pods.
+	if !kubetypes.IsStaticPod(pod) && !kubetypes.IsMirrorPod(pod) {
```

Comment:
Never gets called on a MirrorPod, so the check isn't necessary.

Reply:
Done, checked locally and works fine.